### DISCUSSION TITLE
next/font/google AbortError: The user aborted a request

### DISCUSSION THREAD

**Discussion by ricardo-fnd** [vercel/next.js#55734] [App Router]
I'm getting this error when downloading a font from google, only locally. When testing in `staging` the font is downloaded with success. I'm using next v15

![Screenshot 2023-09-21 at 12 34 22](https://github.com/vercel/next.js/assets/52207061/2592c25b-97f0-487b-84f0-da4d45fad9d0)

**Comment by Dezainer** [comment]
I'm on 13.5.2 and the same issue is happening in my project

![image](https://github.com/vercel/next.js/assets/19918284/fb621a13-3dc4-4e74-904f-2c53a9ca0e4f)

**Comment by Amminn** [comment]
I have the same issue, sometimes the fonts work by themselves and sometimes I do get these errors without changing anything in the layout page

This is the error form the console
```
 ⨯ Failed to download `Raleway` from Google Fonts. Using fallback font instead.

Failed to fetch `Raleway` from Google Fonts.}
 ⨯ Failed to download `Poppins` from Google Fonts. Using fallback font instead.

Failed to fetch `Poppins` from Google Fonts.}
```

This is how I'm importing the font
```javascript
import { Poppins, Raleway } from 'next/font/google';
import './globals.scss'
import Header from './components/Header';
import Footer from './components/Footer';

const raleway = Raleway({
  subsets: ['latin'],
  display: 'swap',
  variable: '--font-raleway',
  weight: ['100', '200', '300', '400', '500', '600', '700', '800', '900'],
});

const poppins = Poppins({
  subsets: ['latin'],
  display: 'swap',
  variable: '--font-poppins',
  weight: ['100', '200', '300', '400', '500', '600', '700', '800', '900'],
});

export const metadata = {
  title: 'Create Next App',
  description: 'Generated by create next app',
}

export default function RootLayout({ children }) {
  return (
    <html lang="en" className={`${poppins.variable} ${raleway.variable}`}>
      <body>
        <Header />
          {children}
        <Footer />
      </body>
    </html>
  )
}
```

Also, note that I'm using SCSS and importing the font as variables like so:
```scss
html {
  font-size: 10px; // 1rem === 10px
  scroll-behavior: smooth;
  font-family: var(--font-poppins);
}
h1, h2, h3 {
  font-family: var(--font-raleway);
}

```

**Comment by Mandy9943** [comment]
Same erro for me, im using pages directory and works well when is deployed but in local never load the fonts
AbortError: The user aborted a request.
    at abort (/home/cesar/projects/beskar/chatbot-ui/node_modules/next/dist/compiled/node-fetch/index.js:1:65190)
    at EventTarget.abortAndFinalize (/home/cesar/projects/beskar/chatbot-ui/node_modules/next/dist/compiled/node-fetch/index.js:1:65410)
    at [nodejs.internal.kHybridDispatch] (node:internal/event_target:757:20)
    at EventTarget.dispatchEvent (node:internal/event_target:692:26)
    at abortSignal (node:internal/abort_controller:369:10)
    at AbortController.abort (node:internal/abort_controller:403:5)
    at Timeout.<anonymous> (/home/cesar/projects/beskar/chatbot-ui/node_modules/next/dist/compiled/@next/font/dist/google/fetch-css-from-google-fonts.js:35:55)
    at listOnTimeout (node:internal/timers:569:17)
    at process.processTimers (node:internal/timers:512:7) {
  type: 'aborted'
}
error - Failed to download `Inter` from Google Fonts. Using fallback font instead.

**Comment by hexionaut** [comment]
I do get the same error ... :)

**Comment by theGobindSingh** [comment]
I am getting the same error on next v13.5 (with pages directory).
A temporary fix is that you **_build your project locally, then run your dev server_**, it should work.
But again, it is not an actual solution.

**Reply by theGobindSingh** [reply]
Update:
You need to delete your .next folder before making the build. So updated steps:

Step 0: delete the .next folder if it exists
Step 1: build your project
Step 2: run your server

😃

**Reply by aqajade** [reply]
forever thanks, brother

**Reply by shreyakjain** [reply]
@theGobindSingh   it's still not working. My builds fail because of not able to download the fonts. I tried deleting .next folder and then do npm run build. But this build keeps failing. Can you pls help with this? I'm using next.js 14.2.10

**Comment by simjieun** [comment]
I'm experiencing the same issue.
I'm getting "aborterror the user aborted a request font" only locally, and I'd appreciate some solutions.

**Reply by theGobindSingh** [reply]
A temporary fix is that you build your project locally, then run your dev server, it should work.

**Reply by Ashley-King** [reply]
Tried this with my fingers crossed but it didn't work for me.

**Reply by theGobindSingh** [reply]
@Ashley-King could you please delete your .next folder, run the build and then run dev server? 
The error won't stop but the font loads 🤷🏻, it has happened to me few times now and it works.

**Reply by Ashley-King** [reply]
Thank you @theGobindSingh! 

I did not delete my .next folder the first time. I've followed your instructions with nextjs 13.5.3 and nextjs 14.0.3 and it DOES work for me. 

The error I was getting was a socket hang up during the google font api request. It retries 3 times and then gives up. Now I'm not getting the error at all. Is the request going through now or is it your code magic?  

Doesn't matter. It works! 😀

Thanks for the help!

**Reply by theGobindSingh** [reply]
@Ashley-King that's great 😃 you're welcome 🫂
I updated my earlier answer as well so it can help others as well 😁

**Comment by aniqueqode** [comment]
I'm also facing similar issue, @aaronbrown-vercel, Please help with this. ‼️

```shell
- error Failed to download `IBM Plex Mono` from Google Fonts. Using fallback font instead.
AbortError: The user aborted a request.
    at abort (/Users/syedanique/Desktop/Qode/platform/node_modules/next/dist/compiled/node-fetch/index.js:1:65190)
```

**Comment by nickali2** [comment]
I get the same error

**Comment by code1379** [comment]
I get the same error

**Comment by philohelp** [comment]
Same error here. The user aborted a request. Did the build trick with no luck. Utterly annoying. I ended up rebooting the project.

**Reply by theGobindSingh** [reply]
Could you please delete your .next folder before making the build and then try?

**Comment by tochiji** [comment]
I'm experiencing the same error.  (`"next": "13.5.4"`)

One thing I can say for certain is that the `AbortError: The user aborted a request` started occurring right after I upgraded my macOS version from `Ventura` to `Sonoma`.

**Comment by tochiji** [comment]
I managed to successfully load the fonts by deleting the `.next` folder and running `npm run dev` (or `next dev`) once more!

**Comment by mo-cherif** [comment]
Just worked for me by deleting the .next and running again the app

**Comment by thatcatfromspace** [comment]
Same problem here. Sometimes it takes deleting the .next folder multiple times to do the trick, especially when you're using several fonts. Hope there'll be a proper fix to it.

**Comment by shahwaiz-kareem** [comment]
This error could be fixed by restarting you dev server, and make sure that you have a good internet connection to download the fonts. If not work just delete .next folder from you project and restart by running `npm run dev`..

**Comment by shitgem** [comment]
Truly weird error. Tried on two different machines and one got this error, but on the second machine everything worked perfectly fine (On the first machine tried deleting the .next folder nad building - didn't work).

**Comment by Moondoyeon** [comment]
same error (next 14.0.4)
i even deleted .next folder -> yarn build -> yarn dev 
but the still getting error 😰

**Comment by humanterror** [comment]
Same error for me...
I've tried to delete the .nextjs folder and build the project but it doesn't work.
I tried to download the fonts locally but got an error for deprecation and the results were the same.
Also, I tried to set NODE_TLS_REJECT_UNAUTHORIZED=0
![image](https://github.com/vercel/next.js/assets/116093262/1672f2ce-05a0-4927-baef-efd16efb0416)

**Comment by nirvikpurkait** [comment]
Maybe it is because of dev server is not able to download the font from google, I don't know if it is correct or not, but  when i get those error I just delete the imported font and use the google instead, _**it is not recomended by Next.Js team**_, you can do this on development, but when building the app you can change those font with the imported font.

**Comment by rrigonia** [comment]
I had the same issue with the `run build` and `run dev` script using node 18 and didnt work with node --lts (v20). Im using the App Route of NextJs v14. 
Just worked for me by adding the 2 links from Google fonts into the <head /> of my root layout

```ts
export default function RootLayout({
	children,
}: Readonly<{
	children: React.ReactNode;
}>) {
	return (
		<html lang="en">
			<head>
                                 // the 2 links provided by Google fonts
				<link rel="preconnect" href="https://fonts.googleapis.com" />
				<link rel="preconnect" href="https://fonts.gstatic.com" crossOrigin="anonymous" />
			</head>
			<body className={inter.className}>
				...
```

**Comment by Wituz** [comment]
This happened to me, because I was on a VPN that for some reason couldn't reach the font. Turning off my VPN resolved the issue.

**Comment by karapy** [comment]
Same error for me...

deleted .next => not working
using vpn => not working
not using vpn => not working
build and then dev => not working
try a couple of time devmode => not working

**Reply by halowahyudi** [reply]
clear cache your browser.

**Comment by hungdoansy** [comment]
I just ran into this issue while it's just fine 1 hour ago before I took a nap

**Reply by hungdoansy** [reply]
I turned off VPN and it worked. I turned it on a few hours later and the issue was gone.

**Reply by halowahyudi** [reply]
clear cache/cookie your browser.

**Comment by ironexdev** [comment]
Same issue, node 20, next 14 (app router) - nothing mentioned above works, except for turning **on** VPN

**Comment by dav-q** [comment]
The same error here, it just happens with some fonts in my case:
**Node**: v20.12.2 LTS

- When I try to use `import { Noto_Serif } from "next/font/google";`, the development and production mode never fails.
    - `next v14.0.3` and `next v14.2.2`

- When I try to use `import { Noto_Serif_SC } from "next/font/google";`, the development mode fails, and the production mode works.

    - with `next v14.0.3` the command `yarn dev` does not work but when I execute `yarn dev --turbo` works for some strange reason, even I can see the font loaded in the `.next` folder 
![image](https://github.com/vercel/next.js/assets/50281001/2600a9a2-fbf3-4abd-aa7e-9a7f64b3de7c)

    - with `next v14.2.2` the commands `yarn dev` and `yarn dev --turbo` does not work
![image](https://github.com/vercel/next.js/assets/50281001/dbfd7383-121a-4e83-8f68-9f0f2046efbd)

**Comment by xuliudong** [comment]
After closing vnp, the problem was solved

**Comment by isneace** [comment]
This is still not working for me, I did the following:
* Cleared browser cache
* `yarn cache clean`
* deleted `.next` directory
* deleted `node_modules`
* re-ran and built everything
* restarted computer
* disabled vpn

None of those things worked.

My specs are:
* next js 14.1.0
* ubuntu 23.04

**Reply by GabeMeister** [reply]
Same.

**Comment by FavoShane** [comment]
any solutions?

**Comment by summerscar** [comment]
i have the same error.

**Comment by violet1107** [comment]
I have the same error too. I've turned off the VPN, the still same error.

**Comment by sholeh007** [comment]
this is webpack issue. you can use turbo to fix this. but when you build, still error.

**Comment by sarsamurmu** [comment]
The problem is that if you look into the font fetcher script at `node_modules/next/dist/compiled/@next/font/dist/google/fetch-font-file.js`, you would see that the timeout of `3000 ms` or `3 s` is applied only on `dev` command, in `build` command there is no such timeout. If your network cannot resolve the request within 3 seconds, it will abort the request for downloading the font files. There is no way to configure the timeout using `next.config.mjs` or other configuration file.

**SOLUTION**: In the mentioned file, just change the timeout to more (like `30000 ms` or `30 s`) and save it. There's no other way with less hassle. You may use [patch-package](https://www.npmjs.com/package/patch-package) to retain the modifications.

BTW, I am using nextjs `14.2.14`

**Comment by navanshu** [comment]
So basically there is no solution, except installing the font from other source

**Comment by abuzar-rasool** [comment]
In my case, i just deleted .next folder and node_modules. Reconnected to my wifi and it worked.

**Comment by thatcatfromspace** [comment]
In a lot of cases, deleting `node_modules` and `.next` directories seems to help, but it's a hit-or-miss as I have to do the delete/install process multiple times sometimes. I went with importing fonts through Google Fonts even though it has a higher overhead. I wanted several fonts in a single component for a project and I had to resort to importing all of the fonts from Google because Next simply refused to import select fonts, which was curious because I was expecting it to fail globally. 

@sarsamurmu's solution works. Seems to be the best monkey patch in this thread.
